docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -11,18 +11,255 @@ Because the cloud and edge is located in the network boundary of different secur
+
+* Cloudgateway and edgegateway are installed and run in the central cloud and each edge site respectively
+* Edgegateway of Edge site can actively establish encrypted links with cloudgateway on the central side, and support two-way message transmission
+* Register the services of edge site and central cloud to cloudgateway
+* Edgegateway in the edge site acts as the service in the central cloud, and exposes the service access interface of the central cloud to the service in the edge site
+* Cloudgateway in the central cloud acts as the service in each edge site and exposes the service access interface of the edge site in the central cloud
+
+### Proposal B
+
+The access of cloud edge components/services is realized based on L3 layer network communication, but the network/IP is not exposed to the remote end, but the components/services are exposed to the remote end.
+
+![](Proposal-B1.png)
+
+Consider providing the concept of virtual presence, which is the local virtual presentation of remote services. In this way, virtual presence can be regarded as a local service with local IP addresses. These IP addresses have nothing to do with the network where the remote service is located, and can be allocated independently. In this way, the problem of unified planning of IP address of VPN solution in the whole network can be solved at the network level. From the perspective of security, virtual presence is completely isolated from the remote service. The client cannot directly access the network where the remote service is located. If the client is attacked, it can not directly attack the server.
+
+![](Proposal-B2.png)
+
+Gateway (cloudgateway/edgegateway) is introduced into the cloud and each edge. Virtual presence of remote service is provided on these gateways, and service components that need communication between cloud and edge site are presented respectively to realize communication access of cloud edge service：
+
+* Edgegateway is installed and run in the central cloud and each edge site respectively, so that these edge sites can access to the central cloud through edgegateway
+* Create edgesite objects to the system and associate edgesite with these edgegateway
+* All components in the cloud and edge that need mutual access are registered with the central cloud cloudgateway in the form of service, and the access policies of these services are configured
+* The central cloud synchronizes the policy to the corresponding edgegateway, which provides virtual presence for the edge through translation mechanisms to simulate local traffic flow
+
```

Comment:
I would suggest that you draw diagrams with example message flows from central to edge and from edge to central to clarify the key difference between option A and option B. From high level, they look kind of similar.

Reply:
Ok, I will add it, thx

Reply:
how does the actual service maintain the mapping with their stub/shadown/phatom or "virtual presence"? let's say if service 3 dies, what happens to its "virtual presence"?

Reply:
also, I still don't quite get what's the benefit of proposal B over proposal A? what does the concept of virtual presence really offer other than adding more complexity of mapping with their real presence.

Reply:
how does the actual service maintain the mapping with their stub/shadown/phatom or "virtual presence"? let's say if service 3 dies, what happens to its "virtual presence"?
-- In this case, nothing will happen to its "virtual presence". The data connection from service 1 to service 3 will be closed(Because the service 3 dies)

Reply:
"virtual presence" is a local virtual presentation of remote services, the benefit of proposal B over proposal A:
a) no man-in-the-middle, no session termination, end-to-end encryption(for example, rest calls(https) between cloud and edge, this communication data is encrypted) 
b) it can easily support some kinds of services(use http、https、kafka、rabbitmq or something else) communications between cloud and edge， if we want to use option A to implement it， we need to implemente based on each protocol we want to support.